### PR TITLE
fix: ignore hidden files for local folder and S3

### PR DESCRIPTION
**Pull request by jupyterjazz**
Fixes both https://github.com/jina-ai/now/issues/814 and https://github.com/jina-ai/now/issues/778

I added a hidden file to a local folder, as well as to s3 bucket, and both tests (for [local folder](https://github.com/jina-ai/now/blob/f00441ea2be423ba6b8042ab08a6681537d9405f/tests/unit/test_detect_schema.py#L32) and [s3 bucket](https://github.com/jina-ai/now/blob/f00441ea2be423ba6b8042ab08a6681537d9405f/tests/unit/test_detect_schema.py#L58)) are passing.

---

- [x] This PR references an open issue
- [x] Test added
- [ ] Documentation adjusted

### CODE REVIEW

File: now/common/detect_schema.py [hunk 1]
```diff
@@ -231,6 +231,10 @@ def set_field_names_from_s3_bucket(user_input: UserInput, **kwargs):
 
     objects = list(bucket.objects.filter(Prefix=folder_prefix))
     file_paths = [obj.key for obj in objects if not obj.key.endswith('/')]
```
Comment (azayz) [inline now/common/detect_schema.py:233]: can you combine conditions in one for loop ?

File: now/common/detect_schema.py [hunk 2]
```diff
@@ -266,8 +270,10 @@ def set_field_names_from_local_folder(user_input: UserInput, **kwargs):
             'The path provided is not a folder, please check documentation https://now.jina.ai'
         )
     file_paths = []
-    for root, dirs, files in os.walk(dataset_path):
-        file_paths.extend([os.path.join(root, file) for file in files])
+    for root, _, files in os.walk(dataset_path):
```
Comment (azayz) [inline now/common/detect_schema.py:273]: can we also try to combine here ?

**Review by violenil — APPROVED** [review]
LGTM!

**Review by azayz — APPROVED** [review]
LGTM

**Pull request merged**